crates/cheatnet/src/cheatcodes/spy_events.rs

```diff
@@ -0,0 +1,69 @@
+use crate::CheatnetState;
+use cairo_felt::Felt252;
+use cairo_vm::hint_processor::hint_processor_utils::felt_to_usize;
+use starknet_api::core::ContractAddress;
+
+/// Represents emitted event. It is used in the `CheatnetState` to keep track of events
+/// emitted in the `cheatnet::src::rpc::call_contract`
+#[derive(Clone)]
+pub struct Event {
+    pub from: ContractAddress,
+    pub name: Felt252,
+    pub keys: Vec<Felt252>,
+    pub data: Vec<Felt252>,
+}
+
+/// Specifies which contract are spied on.
+pub enum SpyOn {
+    All,
+    One(ContractAddress),
```

Comment:
Why to have one target? Isn't it a subset of multiple? Maybe `All` and `Selected` would be better targets.

Reply:
Maybe, but now one can do
```
spy_events(SpyOn::One(contract_address))
```

and then will have to 
```
spy_events(SpyOn::Selected(array![contract_address]))
```

The array is not necessary here